tests/functional/monitoring/conftest.py

```
@@ -307,13 +307,33 @@ def measure_corrupt_pg(request, measurement_dir, threading_lock):
 
+    pool_object = "test_object"
+    ct_pod.exec_ceph_cmd(f"rados -p {pool_name} put {pool_object} /etc/passwd")
+
+    # Find the primary OSD for this PG and use its deployment for corruption.
+    # On clusters with many OSDs the PG may not reside on osd.0, so we must
+    # pick the OSD that actually holds the data.
+    pg_info = ct_pod.exec_ceph_cmd(f"ceph osd map {pool_name} {pool_object}")
+    target_osd_id = pg_info["up"][0]
+    logger.info(
+        f"PG {pg_info['pgid']} primary is osd.{target_osd_id}, "
+        f"selecting its deployment for corruption"
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)